components/server/src/express-util.ts

```diff
@@ -24,9 +24,8 @@ export const query = (...tuples: [string, string][]) => {
 // We control all of those values and the base domain, so we don't need to much effort
 export const isAllowedWebsocketDomain = (originHeader: any, gitpodHostName: string): boolean => {
     if (!originHeader || typeof originHeader !== "string") {
-        return false;
+        return true;
```

Comment:
I think the check here should only return true, if the header is not set. If it's not a string, it should fail (not really sure what other value it could have but oh well..)

This kind of defeats the value of the function entirely. I guess it provides "some" safeguard given the Origin is specified by the browser but.. I'll have a think about this some more.

Reply:
> I guess it provides "some" safeguard given the Origin is specified by the browser []

Exactly that's the point. The purpose of this validation is to disable easy embedding Gitpod into other pages (e.g. via `iframe`) which would allow to trick users and potentially steal their data.

This provides absolutely no security outside of a browser client, i.e. using `websocat` you can easily set a proper origin header and access the WebSocket API.

Reply:
> I think the check here should only return true, if the header is not set. If it's not a string, it should fail (not really sure what other value it could have but oh well..)

I've changed the type of `originHeader` to `string` so we don't need the `typeof` check anymore. Not sure why it had the `any` type - all call sites supply a `string`.